src/transport_ofi.c

```diff
@@ -1433,8 +1433,8 @@ int shmem_transport_startup(void)
                            shmem_internal_params.OFI_STX_MAX);
         }
 
-        if (num_on_node <= 0)
-            RAISE_ERROR_MSG("Invalid number of TX contexts (%d)\n", num_on_node);
+        if (ofi_tx_ctx_cnt <= 0)
```

Comment:
Don't we also need to check that `num_on_node > 0`?  It seems possible for `num_on_node` to be 0 here; for example, if a user enables `OFI_STX_AUTO` but all PEs have unique hostnames.  In that case we'd divide by 0 on the next line, right?

Reply:
Better to rely on the runtime layer to sanity check itself during startup, e.g. in `shmem_runtime_util_populate_node`.  Looks like we should add this check to PMIX.

Reply:
Update runtime-pmix.c in d8cb87e

Reply:
Looks good.  Although, I wasn't sure whether we need to check this in `runtime-mpi.c` as well.  I don't believe I've ever seen a validity check on `MPI_Comm_Size` in practice... but the output is an `int`...